src/store/selectors/deck-filters.ts

```diff
@@ -234,43 +276,61 @@ export const selectDecksFiltered = createSelector(
   },
 );
 
-export const selectFactionsInLocalDecks = createSelector(
-  selectLocalDecks,
-  (state: StoreState) => state.metadata.factions,
-  (decks, factionMeta) => {
-    if (!decks) return [];
-
-    const factionsSet = new Set<string>();
+export const selectDecksSorting = (state: StoreState) => state.deckFilters.sort;
 
-    for (const deck of decks) {
-      factionsSet.add(deck.cards.investigator.card.faction_code);
-    }
+const genericSort = (a: string | number, b: string | number, mod: 1 | -1) => {
```

Comment:
As a convention, we use `function` declarations for top level functions instead of arrows. Reason: they don't require functions to appear in order, so are hoisted.